blog/2023-10-28-netdata-vs-prometheus-performance-analysis/index.md

```diff
@@ -0,0 +1,377 @@
+(<code>disk_percent</code>)</strong>
+   </td>
+   <td><strong>Retention
+(<code>expected_retention</code>)</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Tier 0
+   </td>
+   <td>1.5 TiB
+   </td>
+   <td>100%
+   </td>
+   <td>9.6 days
+   </td>
+  </tr>
+  <tr>
+   <td>Tier 1
+   </td>
+   <td>763 GiB
+   </td>
+   <td>100%
+   </td>
+   <td>43.1 days
+   </td>
+  </tr>
+  <tr>
+   <td>Tier 2
+   </td>
+   <td>381 GiB
+   </td>
+   <td>46%
+   </td>
+   <td>467.5 days
+   </td>
+  </tr>
+</table>
+
+
+Netdata used 1.5 TiB for almost 10 days of high resolution data, while Prometheus used 3.1 TiB for 7 days of high resolution data.
+
+Based on these data, we can estimate the average number of bytes on disk, per data collection sample:
+
+
+
+* Prometheus collects 2.7 million metrics per second. Over the span of 7 days, it accumulated approximately 1.6 trillion samples. Its storage efficiency can be determined as:
+   3.1 TiB / 1.6 trillion samples = **2.13 bytes per sample**.
+   
+   It's worth noting an earlier observation, that there might be occasions when Prometheus doesn't scrape all endpoints every second. This can affect the calculated efficiency If the points collected are significantly less. 
+   Prometheus provides a way to get the typical size of a sample on disk:
+
+   `(rate(prometheus_tsdb_compaction_chunk_size_bytes_sum[7d])) / rate(prometheus_tsdb_compaction_chunk_samples_sum[7d])`
+   
+   In our case, this reports 2.14 bytes per sample, which verifies our estimation.
+
+* Netdata, on the other hand, also collects 2.7 million metrics per second. Over 9.6 days, it ingested about 2.2 trillion samples. The storage efficiency for Netdata is:
+1.5 TiB / 2.2 trillion samples = **0.75 bytes per sample.**
+
+
+Based on these data, Netdata uses 75% less storage than Prometheus, or Prometheus uses 280% more storage than Netdata for the same dataset.
+
+Additionally, it's pertinent to mention that Netdata used the saved storage space to support over a year of data retention, albeit at a reduced resolution.
+
+
+## Disk I/O
+
+
+### Disk Writes
+
+![alt_text](images/image9.png "image_tooltip")
+
+
+_Image: 3 hours of **disk writes**: Netdata is red line, Prometheus is pink line_
+
+* Netdata has a consistent write speed of approximately 2 MiB per second.
+* Prometheus shows a varied write speed, ranging from 20 to 100 MiB per second.
+
+
+### Disk Reads
+
+![alt_text](images/image10.png "image_tooltip")
+
+
+_Image: 3 hours of **disk reads**: Netdata is red line, Prometheus is pink line_
+
+
+
+* Netdata registers minimal to no disk reads.
+* Prometheus, on the other hand, consistently records reads exceeding 50 MiB per second.
+
+From the data, it can be inferred that Netdata primarily writes data directly to their final position, given its steady write speed. Prometheus exhibits variable write and read patterns, possibly suggesting mechanisms like Write-Ahead Logging (WAL) or other data reorganization strategies.
+
+
+## Summary
```

Comment:
@sashwathn should we state the summary stats presented are for 24 hour comparison?
the screenshots above are all 3h or 5min